### ISSUE TITLE
False Positive IsEmptyFunctionUsageInspection

### ISSUE THREAD

**Issue by puggan** [kalessil/phpinspectionsea#1850]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections EA Ultimate 2021.5 (com.kalessil.phpStorm.phpInspectionsUltimate-mp) |
| Language level | PHP 8.0 |

Current behaviour
--
IsEmptyFunctionUsageInspection warning at:
![image](https://user-images.githubusercontent.com/866668/188907571-db949224-b62a-4a17-8112-c271af17055b.png)

```php
<?php

class EaTest
{
    public function mayThrow(): DateTimeImmutable
    {
        if (random_int(0, 1) < 1) {
            throw new \Exception('random fail');
        }
        return new \DateTimeImmutable();
    }

    public function test()
    {
        try {
            $date = $this->mayThrow();
        } catch (\Exception) {
        }

        if (empty($date)) {
            return null;
        }

        return $date;
    }
}
```

Expected behaviour
--
no IsEmptyFunctionUsageInspection, it's ok to use empty as the value is either a DateTimeImutable or undefined.
changing it to `if (!isset($date))` makes more sence then to change it to `if ($date === null)`.

Environment details
---

PhpStorm 2022.2.2 Preview
Build #PS-222.4167.9, built on September 1, 2022
Licensed to {{me}}
Subscription is active until April 24, 2023.
Runtime version: 17.0.4+7-b469.53 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
Windows 11 10.0
GC: G1 Young Generation, G1 Old Generation
Memory: 2010M
Cores: 8
Registry:
    run.processes.with.pty=TRUE

Non-Bundled Plugins:
    com.atlassian.bitbucket.references (2020.3.206)
    com.jetbrains.plugins.jade (222.4167.4)
    com.kalessil.phpStorm.phpInspectionsUltimate-mp (2021.5)
    NEON support (0.5.2)
    ru.adelf.idea.dotenv (2022.2)
    de.espend.idea.php.toolbox (6.1.0)
    de.espend.idea.php.annotation (8.2.3)
    aws.toolkit (1.50-222)
    izhangzhihao.rainbow.brackets (6.25)
    fr.adrienbrault.idea.symfony2plugin (2022.1.234)
